packages/caliper-core/lib/manager/monitors/monitor-interface.js

```diff
@@ -15,19 +15,20 @@
 
 'use strict';
 
+const ConfigUtil = require('../../common/config/config-util.js');
+
 // TODO: now we record the performance information in local variable, it's better to use db later
 /**
  * Interface of resource consumption monitor
  */
 class MonitorInterface{
     /**
      * Constructor
-     * @param {JSON} monitorConfig Configuration object for the monitor
-     * @param {number} interval Watching interval, in seconds
+     * @param {JSON} resourceMonitorOptions Configuration options for the monitor
      */
-    constructor(monitorConfig, interval) {
-        this.monitorConfig = monitorConfig;
-        this.interval     = interval*1000; // convert to ms
+    constructor(resourceMonitorOptions) {
+        this.options = resourceMonitorOptions;
+        this.interval = resourceMonitorOptions.interval ? resourceMonitorOptions.interval*1000 : ConfigUtil.get(ConfigUtil.keys.Progress.Reporting.Interval);
```

Comment:
I'd leave this to the specific monitor implementations. But since a "polling interval" is probably common for all resource monitors, it can also stay here (but I wouldn't default to the progress report setting, it has different semantics).